lib/dockerregistry/inventory.go

```diff
@@ -51,59 +54,373 @@ func getSrcRegistry(rcs []RegistryContext) (*RegistryContext, error) {
-func ParseManifestFromFile(
-	filePath string) (Manifest, error) {
+func ParseManifestFromFile(filePath string) (Manifest, error) {
 
 	var mfest Manifest
-	var rd RenamesDenormalized
-	var srcRegistry *RegistryContext
+	var empty Manifest
+
 	b, err := ioutil.ReadFile(filePath)
 	if err != nil {
-		return mfest, err
+		return empty, err
 	}
 
 	mfest, err = ParseManifestYAML(b)
 	if err != nil {
-		return mfest, err
+		return empty, err
 	}
 
 	mfest.filepath = filePath
 
+	err = mfest.finalize()
+	if err != nil {
+		return empty, err
+	}
+
+	return mfest, nil
+}
+
+func (m *Manifest) finalize() error {
 	// Perform semantic checks (beyond just YAML validation).
-	srcRegistry, err = getSrcRegistry(mfest.Registries)
+	srcRegistry, err := getSrcRegistry(m.Registries)
 	if err != nil {
-		return mfest, err
+		return err
 	}
-	mfest.srcRegistry = srcRegistry
+	m.srcRegistry = srcRegistry
 
-	rd, err = DenormalizeRenames(mfest, srcRegistry.Name)
+	rd, err := DenormalizeRenames(*m, srcRegistry.Name)
 	if err != nil {
-		return mfest, err
+		return err
 	}
-	mfest.renamesDenormalized = rd
+	m.renamesDenormalized = rd
 
-	return mfest, nil
+	return nil
+}
+
+// ParseManifestsFromDir parses all Manifest files within a directory. We
+// effectively have to create a map of manifests, keyed by the source registry
+// (there can only be 1 source registry).
+func ParseManifestsFromDir(dir string) ([]Manifest, error) {
+	mfests := make([]Manifest, 0)
+
+	var parseAsManifest filepath.WalkFunc = func(path string,
+		info os.FileInfo,
+		err error) error {
+
+		if err != nil {
+			// Prevent panic in case of incoming errors accessing this path.
+			klog.Errorf("failure accessing a path %q: %v\n", path, err)
+		}
+
+		// Skip directories (because they are not YAML files).
+		if info.IsDir() {
+			return nil
+		}
+
+		// First try to parse the path as a manifest file. The only requirement
+		// is that the file must end with a ".yaml" extension. We can be more
+		// restrictive in the future (maybe it will be all files with a certain
+		// pattern, passable from the CLI as an option), but let's cross that
+		// bridge when we get there.
+
+		// Skip non-YAML files.
+		if !strings.HasSuffix(path, ".yaml") {
+			return nil
+		}
+
+		mfest, errParse := ParseManifestFromFile(path)
+		if errParse != nil {
+			klog.Errorf("could not parse manifest file '%s'\n", path)
+			return errParse
+		}
+
+		// Save successful parse result.
+		mfests = append(mfests, mfest)
+
+		return nil
+	}
+
+	if err := filepath.Walk(dir, parseAsManifest); err != nil {
+		return mfests, err
+	}
+
+	if len(mfests) == 0 {
+		return nil, fmt.Errorf("no manifests found in dir: %s", dir)
+	}
+
+	return mfests, nil
+}
+
+// ValidateManifestsFromDir parses checks for errors found in the manifests,
+// taken as a whole.
+//
+// nolint[gocyclo]
+func ValidateManifestsFromDir(mfests []Manifest) error {
+	if len(mfests) == 0 {
+		return fmt.Errorf("no manifests to validate")
+	}
+
+	if len(mfests) == 1 {
```

Comment:
I don't think it is worth skipping this validation here - even if it shouldn't catch anything today.

Reply:
Can I fix this in a new PR?